### ISSUE TITLE
Export Azure HDInsight Cluster ID

### ISSUE THREAD

**Issue by kabal2010** [hashicorp/terraform-provider-azurerm#26180]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Currently, the Azure HD Insight Cluster ID is not part of the variables exported with a data pull or after deployment

### New or Affected Resource(s)/Data Source(s)

azurerm_hdinsight_hadoop_cluster, azurerm_hdinsight_hbase_cluster, azurerm_hdinsight_interactive_query_cluster, azurerm_hdinsight_kafka_cluster, azurerm_hdinsight_spark_cluster

### Potential Terraform Configuration

```hcl
data "azurerm_hdinsight_cluster" "example" {
  name                = "example"
  resource_group_name = "example-resources"
}

output "cluster_id" {
  value = data.azurerm_hdinsight_cluster.example.cluster_id
}
```

**Comment by Chambras** [comment]
Hi @kabal2010 
I was looking at the code and I did some testing and actually you do get the ID and the name. This is the code I tried:
```hcl
data "azurerm_hdinsight_cluster" "example" {
  name                = "example"
  resource_group_name = "example-resources"
}

output "cluster_id" {
  value = data.azurerm_hdinsight_cluster.example.id
}

output "name" {
  value = data.azurerm_hdinsight_cluster.example.name
}
```

The documentation might out of date. I will submit a PR to update the documentation.

**Comment by kabal2010** [comment]
@Chambras 
What I mean is *`cluster id`* and not the *`resource id`*. Cluster ID is dynamically generated for the creation of the resources. The output below is for the *`resource id`* and not *`cluster id`*
```hcl
output "cluster_id" {
  value = data.azurerm_hdinsight_cluster.example.id
}
```

**Comment by Chambras** [comment]
@kabal2010 I see your point now. Trying to get a PR ready for this week :)

**Comment by kabal2010** [comment]
@Chambras 
That's excellent. Will wait for the completion and test afterwards to confirm.

**Comment by kabal2010** [comment]
@Chambras Any update with regards to this request?

**Comment by Chambras** [comment]
@kabal2010 I submitted this PR https://github.com/hashicorp/terraform-provider-azurerm/pull/26228
You can ask for status there and like it so the reviewers see it :) 
It seems all tests/checks passed so hopefully it will be approved and merged soon :)

**Comment by Chambras** [comment]
@kabal2010 It is finally merged! and part of 3.109.0 go get it while it is still hot :) 
Let me know how it goes!

**Comment by kabal2010** [comment]
@Chambras Already working on testing. Should be able to give you feedback in a couple of days.
